gcc 5 (experimental C++14 support) and boost-1.57

```
/c/bitmonero/contrib/epee/include/storages/portable_storage.h:430:16: warning: converting to ‘bool’ from ‘std::nullptr_t’ requires direct-initialization [-fpermissive]
         return nullptr;
                ^
```

```
/c/bitmonero/src/p2p/net_node.inl:1108:370: error: redeclaration of ‘const peerid_type pr’
In file included from /c/bitmonero/src/p2p/net_node.h:254:0,
                 from /c/bitmonero/src/daemon/daemon.cpp:44:
/c/bitmonero/src/p2p/net_node.inl:1106:37: note: ‘const peerid_type pr’ previously declared here
         if(rsp.status != PING_OK_RESPONSE_STATUS_TEXT || pr != rsp.peer_id)
                                     ^
```

```
/c/bitmonero/src/p2p/net_node.inl:1098:9: sorry, unimplemented: non-trivial designated initializers not supported
         [=](int code, const COMMAND_PING::response& rsp, p2p_connection_context& context)
         ^
```

--- a/contrib/epee/include/storages/portable_storage.h
+++ b/contrib/epee/include/storages/portable_storage.h
@@ -427,7 +427,7 @@ namespace epee
       TRY_ENTRY();
       CHECK_AND_ASSERT(hsec_array, false);
       if(hsec_array->type() != typeid(array_entry_t<section>))
-        return nullptr;
+        return false;
       array_entry_t<section>& sec_array = boost::get<array_entry_t<section>>(*hsec_array);
       h_child_section = sec_array.get_next_val();
       if(!h_child_section)

--- a/src/p2p/net_node.inl
+++ b/src/p2p/net_node.inl
@@ -1194,6 +1194,10 @@ namespace nodetool
       std::string port_=port;
       peerid_type pr_ = pr;
       auto cb_ = cb;*/
+
+      // GCC 5.1.0 gives error with second use of uint64_t (peerid_type) variable.
+      peerid_type pr_ = pr;
+
       bool inv_call_res = epee::net_utils::async_invoke_remote_command2<COMMAND_PING::response>(ping_context.m_connection_id, COMMAND_PING::ID, req, m_net_server.get_config_object(),
         [=](int code, const COMMAND_PING::response& rsp, p2p_connection_context& context)
       {
@@ -1205,7 +1209,7 @@ namespace nodetool
 
         if(rsp.status != PING_OK_RESPONSE_STATUS_TEXT || pr != rsp.peer_id)
         {
-          LOG_PRINT_CC_L2(ping_context, "back ping invoke wrong response \"" << rsp.status << "\" from" << ip << ":" << port << ", hsh_peer_id=" << pr << ", rsp.peer_id=" << rsp.peer_id);
+          LOG_PRINT_CC_L2(ping_context, "back ping invoke wrong response \"" << rsp.status << "\" from" << ip << ":" << port << ", hsh_peer_id=" << pr_ << ", rsp.peer_id=" << rsp.peer_id);
           return;
         }
         m_net_server.get_config_object().close(ping_context.m_connection_id);

